### ISSUE TITLE
json body POST 

### ISSUE THREAD

**Issue by carlskii** [stajkowski/webhooks#4]
Hi,

I've been doing some more testing and think I've worked out the reason I'm getting 400 responses from server when I do a post. 

I set up a basic echo server to watch what was being posted. This what should be sent, I used a REST client to make the request here: 

```json
{
  "ip": "127.0.0.1",
  "time": 1425557849126,
  "method": "POST",
  "url": "/vedsdk/authorize/",
  "body": "{\n  \"Username\": \"admin\",\n  \"Password\": \"Passw0rd\"\n}",
  "headers": {
    "content-type": "application/json",
    "host": "127.0.0.1:5000",
    "connection": "close",
    "user-agent": "Paw/2.1.1 (Macintosh; OS X/10.10.2) GCDHTTPRequest",
    "content-length": "51"
  },
  "meta": {
    "num": 9,
    "live": 1,
    "status": 200,
    "delay": 0
  }
}
```

This is what the webhooks cookbook is sending: 

```json
{
  "ip": "127.0.0.1",
  "time": 1425557863264,
  "method": "POST",
  "url": "/",
  "body": "Username=admin&Password=Passw0rd",
  "headers": {
    "accept-encoding": "gzip;q=1.0,deflate;q=0.6,identity;q=0.3",
    "accept": "*/*",
    "user-agent": "Ruby",
    "content-type": "application/json",
    "host": "127.0.0.1:5000",
    "content-length": "32"
  },
  "meta": {
    "num": 10,
    "live": 1,
    "status": 200,
    "delay": 0
  }
}
```

So the json for the body `"body": "Username=admin&Password=Passw0rd",` is not created as it should be. Heres my webhooks block:

```ruby
webhooks_request "Get TPP API Key" do
  uri "127.0.0.1/"
  uri_port 5000
  ssl_validation false
  use_ssl false
  follow_redirect true
  post_data (
                { 'Username' => 'admin', 'Password' => 'Passw0rd'}
            )
  header_data (
                  { 'content-type' => 'application/json'}
              )
  expected_response_codes [ 200, 201, 301 ]
  action :post
end
```

So it seems that it could be ignoring the `"content-type": "application/json"` header.

**Comment by carlskii** [comment]
So I worked out what was happening here and changed this line in the provider file:

```ruby
#req.set_form_data( @current_resource.post_data )  #set the form data
req.body = JSON.dump(@current_resource.post_data) #set the body data
```
Which obviously sends it as JSON instead of form data. 

I think it would be worth trying to deal with this using the attributes.

**Comment by stajkowski** [comment]
Hey!

Yes, this is a good find.  The original intent was just to pass the data as a typical POST, seems you have found one shortcoming.  I'll add the ability to send JSON and even XML data as well.  Thanks for the find and a great addition to this cookbook!

**Comment by stajkowski** [comment]
Doing some major work on this cookbook.  I should be able to get this in by Monday.

**Comment by stajkowski** [comment]
Fixed in v0.1.3
